incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/openapi/http/proxy/OpenApiHttpProxyConfigGenerator.java

```diff
@@ -0,0 +1,519 @@
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import jakarta.json.Json;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonPatch;
+import jakarta.json.bind.Jsonb;
+import jakarta.json.bind.JsonbBuilder;
+import jakarta.json.spi.JsonProvider;
+
+import io.aklivity.zilla.runtime.binding.http.config.HttpConditionConfig;
+import io.aklivity.zilla.runtime.binding.http.config.HttpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.http.config.HttpOptionsConfigBuilder;
+import io.aklivity.zilla.runtime.binding.tcp.config.TcpConditionConfig;
+import io.aklivity.zilla.runtime.binding.tcp.config.TcpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.tls.config.TlsOptionsConfig;
+import io.aklivity.zilla.runtime.command.config.internal.airline.ConfigGenerator;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model.OpenApi;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model.Server;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model2.PathItem2;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model2.Server2;
+import io.aklivity.zilla.runtime.engine.config.BindingConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.ConfigWriter;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.RouteConfigBuilder;
+import io.aklivity.zilla.runtime.guard.jwt.config.JwtOptionsConfig;
+import io.aklivity.zilla.runtime.vault.filesystem.config.FileSystemOptionsConfig;
+
+public class OpenApiHttpProxyConfigGenerator implements ConfigGenerator
+{
+    private final OpenApi openApi;
+    private final int[] allPorts;
+    private final int[] httpPorts;
+    private final int[] httpsPorts;
+    private final boolean isPlainEnabled;
+    private final boolean isTlsEnabled;
+    private final Map<String, String> securitySchemes;
+    private final boolean isJwtEnabled;
+    private final NamespaceConfig namespace;
+    private final JsonPatch envVarsPatch;
+    private final ConfigWriter configWriter;
+
+    public OpenApiHttpProxyConfigGenerator(
+        InputStream inputStream)
+    {
+        this.openApi = parseOpenApi(inputStream);
+        this.allPorts = resolveAllPorts();
+        this.httpPorts = resolvePortsForScheme("http");
+        this.httpsPorts = resolvePortsForScheme("https");
+        this.isPlainEnabled = httpPorts != null;
+        this.isTlsEnabled = httpsPorts != null;
+        this.securitySchemes = resolveSecuritySchemes();
+        this.isJwtEnabled = !securitySchemes.isEmpty();
+        this.namespace = createNamespace();
+        this.envVarsPatch = createEnvVarsPatch();
+        this.configWriter = new ConfigWriter(null);
```

Comment:
Recommend we do the majority of the work in the `generate` method instead of in the constructor.